layers/meta-balena-raspberrypi/recipes-kernel/linux/linux-raspberrypi/0007-ovl-fix-regression-caused-by-overlapping-layers-detection.patch

```diff
@@ -0,0 +1,229 @@
+From 9c13e770cca978b8d5c870f2270cdf3a10c63817 Mon Sep 17 00:00:00 2001
+From: Amir Goldstein <amir73il@gmail.com>
+Date: Fri, 12 Jul 2019 15:24:34 +0300
+Subject: [PATCH] ovl: fix regression caused by overlapping layers detection
+
+commit 0be0bfd2de9dfdd2098a9c5b14bdd8f739c9165d upstream.
+
+Once upon a time, commit 2cac0c00a6cd ("ovl: get exclusive ownership on
+upper/work dirs") in v4.13 added some sanity checks on overlayfs layers.
+This change caused a docker regression. The root cause was mount leaks
+by docker, which as far as I know, still exist.
+
+To mitigate the regression, commit 85fdee1eef1a ("ovl: fix regression
+caused by exclusive upper/work dir protection") in v4.14 turned the
+mount errors into warnings for the default index=off configuration.
+
+Recently, commit 146d62e5a586 ("ovl: detect overlapping layers") in
+v5.2, re-introduced exclusive upper/work dir checks regardless of
+index=off configuration.
+
+This changes the status quo and mount leak related bug reports have
+started to re-surface. Restore the status quo to fix the regressions.
+To clarify, index=off does NOT relax overlapping layers check for this
+ovelayfs mount. index=off only relaxes exclusive upper/work dir checks
+with another overlayfs mount.
+
+To cover the part of overlapping layers detection that used the
+exclusive upper/work dir checks to detect overlap with self upper/work
+dir, add a trap also on the work base dir.
+
+Link: https://github.com/moby/moby/issues/34672
+Link: https://lore.kernel.org/linux-fsdevel/20171006121405.GA32700@veci.piliscsaba.szeredi.hu/
+Link: https://github.com/containers/libpod/issues/3540
+Fixes: 146d62e5a586 ("ovl: detect overlapping layers")
+Cc: <stable@vger.kernel.org> # v4.19+
+Signed-off-by: Amir Goldstein <amir73il@gmail.com>
+Tested-by: Colin Walters <walters@verbum.org>
+Signed-off-by: Miklos Szeredi <mszeredi@redhat.com>
+Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
+---
```

Comment:
You should add here Signed-off-by and Upstream-Status such as here for example: https://github.com/balena-os/balena-raspberrypi/blob/master/layers/meta-balena-raspberrypi/recipes-kernel/linux/linux-raspberrypi/0002-wireless-wext-Bring-back-ndo_do_ioctl-fallback.patch#L8-L10

Reply:
what should I put here?
we're waiting to get this patch from meta-raspberrypi...

Reply:
You need to add the final 'Signed-off-by' that mentions your name, and before the first 'Signed-off-by' there should be an 
Upstream-status: <actual status>
The actual status could be Pending/Backport/Inappropriate [configuration]/etc

Reply:
@robertgzr what do you mean we are waiting to get the patch from meta-raspberrypi?

Reply:
@floion  I'm just applying https://github.com/raspberrypi/linux/commit/9c13e770cca978b8d5c870f2270cdf3a10c63817

Reply:
once we update meta-raspberry to a release that has this we can remove this patch again

Reply:
I guess I should do: `Upstream-status: Backport` then...

Reply:
Yes, add it as Backport and also set Signed off by yourself there as in the example and as Alex detailed